Antidote Verifier/Sdl.Community.AntidoteVerifier/Sdl.Community.AntidoteVerifier/Utils/CustomTextCollectionVisitor.cs

```diff
@@ -0,0 +1,333 @@
+        }
+
+        //check if the markup is in the list of the markups visited
+        private bool ListContainsMarkupData(IAbstractMarkupData theMarkupData)
+        {
+            bool contains = false;
+            //Why isn't the text in a PlaceHolder in the segment?
+            Location newItemLocation = IsAncestorOfMarkup(theMarkupData, _segment) ? new Location(_segment, theMarkupData) : new Location(theMarkupData.Parent, theMarkupData);
+
+            foreach (IAbstractMarkupData item in _markupsListVisited)
+            {
+                if (item.Equals(theMarkupData))
+                {
+                    Location itemLocation = IsAncestorOfMarkup(theMarkupData, _segment) ? new Location(_segment, item) : new Location(item.Parent, item);
+                    if (newItemLocation != null && itemLocation != null)
+                    {
+                        contains = itemLocation.Equals(newItemLocation);
+                    }
+                }
+            }
+            return contains;
+        }
+
+        //visit functions
+        public void VisitCommentMarker(ICommentMarker commentMarker)
+        {
+        }
+
+        public void VisitLocationMarker(ILocationMarker location)
+        {
+        }
+
+        public void VisitLockedContent(ILockedContent lockedContent)
+        {
+            // all locked content must be counted, as it is treated
+            // as a single unit.
+            if (_inLockedContent)
+            {
+                // we are already iterating recursively through all locked content
+                // must not do it again here, as that would cause nested locked
+                // content to be counted multiple times.
+                return;
+            }
+            _inLockedContent = true;
+            foreach (IAbstractMarkupData item in lockedContent.Content.AllSubItems)
+            {
+                item.AcceptVisitor(this);
+            }
+            _inLockedContent = false;
+        }
+
+        public void VisitOtherMarker(IOtherMarker marker)
+        {
+        }
+
+        public void VisitPlaceholderTag(IPlaceholderTag tag)
+        {
+            if (tag.Properties.DisplayText.Length > 0)
+            {
+                RangeOfCharacterInfos item = new RangeOfCharacterInfos { start = _collectedText.Length, length = tag.Properties.DisplayText.Length };
+                _listOfLockedRanges.Add(item);
+                _collectedText += tag.Properties.DisplayText;
+            }
+        }
+
+        public void VisitRevisionMarker(IRevisionMarker revisionMarker)
+        {
+            //don't add delete revisions
+            if (revisionMarker.Properties.RevisionType != RevisionType.Delete)
+            {
+                foreach (IAbstractMarkupData item in revisionMarker.AllSubItems)
+                {
+                    item.AcceptVisitor(this);
+                }
+            }
+        }
+
+        public void VisitSegment(ISegment segment)
```

Comment:
In the examples used to modify the connector, there were no embedded segments. So, this type of structure was not considered. The function VisitSegment() should be implemented to support these structures, probably in this way :
    public void VisitSegment(ISegment segment)
    {
          foreach (IAbstractMarkupData item in segment.AllSubItems)
            {
                item.AcceptVisitor(this);
            }
    }